### ISSUE TITLE
 [Editor] Allow the user to add viewports in the editor

### ISSUE THREAD

**Issue by setzer22** [urho3d/Urho3D#768]
We just noticed in: http://urho3d.prophpbb.com/post7005.html#p7005 that the editor code doesn't take into account any defined viewports besides its own. And thus, creating new viewports, either to extend the editor's functionality or just to allow for game behaviour is not working as they get removed.

The responsible lines for this are in EditorView.as, function UpdateCameraPreview. Specifically: 

   // Remove extra viewport if it exists and no camera is selected
    if (previewCamera.Get() is null)
    {
        if (renderer.numViewports > viewports.length)
            renderer.numViewports = viewports.length;
    }

There should be a mechanism for either the editor noticing user-created viewports or the user adding his viewports to the editor so it won't delete them.

**Comment by cadaver** [comment]
This is a good suggestion, but in general I'm afraid you'll be fighting a losing battle if you want to run live game code within the editor and keep both the editor and the game stepping on each others' toes. E.g. there is only one UI hierarchy in the engine, and if the game tries to take control of it, it doesn't spell good for the editor.

For the most part, the script-based editor that comes with Urho is a scene editor, and not a live game preview like Unity. To do the latter robustly, the editor and game should not share any usage of the Urho subsystems. There are some external editor projects discussed on the Urho forum that achieve that by using native UI toolkits, but for the built-in editor that's out of scope.

**Comment by friesencr** [comment]
Letting a viewport use an in scene camera isn't out of scope.  There are usability issues with changing the editor camera to the scene stuff.  The editor doesn't know if it shouldn't mutate the camera so it will happily move the camera without your permission.  I was the author of the multiple viewports stuff and was going to add this capability when I wrote the original.  I had started writing it but I forgot why I stopped....
